python/lsst/ts/MTAlignment/mock/mock_t2sa.py

```diff
@@ -0,0 +1,926 @@
+                ("?POS M1M3", self.write_point_group_position_m1m3, ""),
+                ("?POS M2", self.write_point_group_position_m2, ""),
+                ("?STAT", self.write_status, ""),
+            )
+        }
+
+        self.collection_point_regexp = re.compile(r"(?P<group>.*)_P(?P<index>.*)")
+
+        duplicate_keys = self.dispatchers.keys() & self.canned_replies.keys()
+        if duplicate_keys:
+            raise RuntimeError(
+                f"Bug: keys {duplicate_keys} appear in both "
+                "canned_replies and command_handlers"
+            )
+
+        self._laser_warmup_start_tai: None | float = None
+        self._reference_group = "M1M3"
+        # TODO: Get valid working frame values.
+        self.valid_working_frame = {""}
+        # TODO: Get good default working frame value.
+        self.working_frame = ""
+        self.laser_status = "LOFF"
+        self.laser_warmup_time = 60.0
+
+        self.t2sa_status = "READY"
+
+        # How long to wait while pretending to measure (seconds)
+        self.measurement_duration = 2.0
+
+        # Define the "bodies" in the system. Assume m1m3 is at the origin of
+        # the coordinate frame pointing up, m2 is 3 meters away, pointing down
+        # and camera is 2 meters away pointing down.
+        self.position_optimum = dict(
+            m1m3=MockT2SAPointGroup(
+                origin=CartesianCoordinate(0.0, 0.0, 0.0),
+                rotation=BodyRotation(0.0, 0.0, 0.0),
+                radius=8.40,
+            ),
+            m2=MockT2SAPointGroup(
+                origin=CartesianCoordinate(0.0, 0.0, 3.0),
+                rotation=BodyRotation(0.0, 0.0, 0.0),
+                radius=1.74,
+            ),
+            cam=MockT2SAPointGroup(
+                origin=CartesianCoordinate(0.0, 0.0, 2.0),
+                rotation=BodyRotation(0.0, 0.0, 0.0),
+                radius=0.85,
+            ),
+        )
+
+        # Randomize position by 1mm and rotation by (approx) 20 arcsec.
+        self.position_current = dict(
+            m1m3=MockT2SAPointGroup(
+                origin=CartesianCoordinate(*np.random.normal(0.0, 1e-3, 3)),
+                rotation=BodyRotation(*np.random.normal(0.0, 6e-3, 3)),
+                radius=8.40,
+            ),
+            m2=MockT2SAPointGroup(
+                origin=CartesianCoordinate(
+                    *(np.array([0.0, 0.0, 3.0]) + np.random.normal(0.0, 1e-3, 3))
+                ),
+                rotation=BodyRotation(*np.random.normal(0.0, 6e-3, 3)),
+                radius=1.74,
+            ),
+            cam=MockT2SAPointGroup(
+                origin=CartesianCoordinate(
+                    *(np.array([0.0, 0.0, 2.0]) + np.random.normal(0.0, 1e-3, 3))
+                ),
+                rotation=BodyRotation(*np.random.normal(0.0, 6e-3, 3)),
+                radius=0.85,
+            ),
+        )
+
+        self._tel_az = 90.0
+        self._tel_alt = 0.0
+        self._tel_rot = 0.0
```

Comment:
Please add a code comment explaining what these are. Also, please consider a dataclass to hold these values, so they are kept together in a single variable.